lib/azure/armrest/armrest_manager.rb

```diff
@@ -389,10 +392,32 @@ def rest_delete(url)
       end
 
       # Take an array of URI elements and join the together with the API version.
-      def url_with_api_version(*paths)
+      def url_with_api_version(api_version, *paths)
         File.join(*paths) << "?api-version=#{api_version}"
       end
 
+       # Build a one-time lookup table for each provider & resource. This
+        # lets subclasses set api-version strings properly for each method
+        # depending on whichever provider they're using.
+        #
+        # e.g. @@providers['Microsoft.Compute']['virtualMachines']['api_version']
+        #
+        # Note that for methods that don't depend on a resource type should use
+        # the @@api_version class variable instead or set it explicitly as needed.
+        #
+      def set_providers_info
+        if @@providers.empty? && @token
```

Comment:
return earlier if condition not met, and keep the main logic in the main body without a conditional block